cmd/agent/app/reporter/http/reporter.go

```diff
@@ -0,0 +1,236 @@
+				collectorHostPorts[i] = u.Host
+				scheme = u.Scheme
+
+				// and let the user know that we are changing the value they specied!
+				// they might have specified a path and we are ignoring it
+				zlogger.Info(`the "collectorHostPorts" was expected to have only hostname:port.`, zap.String("collectorHostPorts", c))
+			}
+		}
+	}
+
+	return &Reporter{
+		scheme:             scheme,
+		collectorHostPorts: collectorHostPorts,
+
+		authToken: authToken,
+		username:  username,
+		password:  password,
+
+		logger:         zlogger,
+		batchesMetrics: batchesMetrics,
+	}, nil
+}
+
+// EmitZipkinBatch implements EmitZipkinBatch() of Reporter
+func (r *Reporter) EmitZipkinBatch(spans []*zipkincore.Span) error {
+	submissionFunc := func() error {
+		t := thrift.NewTMemoryBuffer()
+		p := thrift.NewTBinaryProtocolTransport(t)
+		p.WriteListBegin(thrift.STRUCT, len(spans))
+		for _, s := range spans {
+			s.Write(p)
+		}
+		p.WriteListEnd()
+		body := t.Buffer.Bytes()
+
+		url := r.Endpoint() + `/api/v1/spans`
+
+		// this can't fail: both the HTTP and the URL method will always be valid
+		// the only part of the URL that can be invalid is the hostname, but we'll fail
+		// early on if it's invalid
+		req, _ := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(body))
+
+		req.Header.Add("Content-Type", "application/x-thrift")
+		r.addAuthHeaders(req)
+
+		res, err := httpClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		if res.StatusCode > 299 { // we are assuming that redirects are bad
+			return fmt.Errorf("failed to submit batch: %s", res.Status)
+		}
+
+		return nil
+	}
+
+	return r.submitAndReport(
+		submissionFunc,
+		"Could not submit zipkin batch",
+		int64(len(spans)),
+		r.batchesMetrics[zipkinBatches],
+	)
+}
+
+// EmitBatch implements EmitBatch() of Reporter
+func (r *Reporter) EmitBatch(batch *jaeger.Batch) error {
+	submissionFunc := func() error {
+		tser := thrift.NewTSerializer()
+		body, _ := tser.Write(batch)
+
+		url := r.Endpoint() + `/api/traces?format=jaeger.thrift`
+
+		// this can't fail: both the HTTP and the URL method will always be valid
+		// the only part of the URL that can be invalid is the hostname, but we'll fail
+		// early on if it's invalid
+		req, _ := http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(body)))
+
+		r.addAuthHeaders(req)
+
+		res, err := httpClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		if res.StatusCode > 299 { // we are assuming that redirects are bad
+			return fmt.Errorf("failed to submit batch: %s", res.Status)
+		}
+
+		return nil
+	}
+
+	return r.submitAndReport(
+		submissionFunc,
+		"Could not submit jaeger batch",
+		int64(len(batch.Spans)),
+		r.batchesMetrics[jaegerBatches],
+	)
+}
+
+// Endpoint returns the endpoint used when communicating with the remote collector
+func (r *Reporter) Endpoint() string {
+	// TODO: do we want to do client-side load balancing? Or use a retry logic?
```

Comment:
yes we do, if we care about performance. TChannel always sends to less-busy connection internally.

We could use LB from go-kit.

Reply:
Would it be OK to have it in a next iteration, or is it required for this one? I would then create an issue and assign to myself.